astronomer/providers/snowflake/hooks/snowflake_sql_api.py

```diff
@@ -0,0 +1,241 @@
+            )
+
+    def execute_query(
+        self, sql: str, statement_count: int, query_tag: str = "", bindings: Optional[Dict[str, Any]] = None
+    ) -> List[str]:
+        """
+        Using SnowflakeSQL API, run the query in snowflake by making API request
+
+        :param sql: the sql string to be executed with possibly multiple statements
+        :param statement_count: set the MULTI_STATEMENT_COUNT field to the number of SQL statements in the request
+        :param query_tag: (Optional) Query tag that you want to associate with the SQL statement.
+            For details, see https://docs.snowflake.com/en/sql-reference/parameters.html#label-query-tag parameter.
+        :param bindings: (Optional) Values of bind variables in the SQL statement.
+            When executing the statement, Snowflake replaces placeholders (? and :name) in
+            the statement with these specified values.
+        """
+        conn_config = self._get_conn_params()
+
+        req_id = uuid.uuid4()
+        url = "https://{0}.snowflakecomputing.com/api/v2/statements".format(conn_config["account"])
+        params: Optional[Union[Dict[str, Any]]] = {"requestId": str(req_id), "async": True, "pageSize": 10}
+        headers = self.get_headers()
+        if bindings is None:
+            bindings = {}
+        data = {
+            "statement": sql,
+            "timeout": 60,
+            "resultSetMetaData": {"format": "json"},
+            "database": conn_config["database"],
+            "schema": conn_config["schema"],
+            "warehouse": conn_config["warehouse"],
+            "role": conn_config["role"],
+            "bindings": bindings,
+            "parameters": {
+                "MULTI_STATEMENT_COUNT": statement_count,
+                "query_tag": query_tag,
+            },
+        }
+        response = requests.post(url, json=data, headers=headers, params=params)
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:  # pragma: no cover
+            raise AirflowException(
+                f"Response: {e.response.content}, " f"Status Code: {e.response.status_code}"
+            )  # pragma: no cover
+        json_response = response.json()
+        self.log.info("Snowflake SQL POST API response: %s", json_response)
+        if "statementHandles" in json_response:
+            self.query_ids = json_response["statementHandles"]
+        elif "statementHandle" in json_response:
+            self.query_ids.append(json_response["statementHandle"])
+        else:
+            raise AirflowException("No statementHandle/statementHandles present in response")
+        return self.query_ids
+
+    def get_headers(self) -> Dict[str, Any]:
+        """Based on the private key, and with connection details JWT Token is generated and header is formed"""
+        if not self.private_key:
+            self.get_private_key()
```

Comment:
line 160 should have a private key. you should be able to access it with `conn_config [private_key]` so you can first call _get_conn_params and then access private I believe

Reply:
`conn_config [private_key] ` is decode in different format I wanted the exact private key so that it is JWTGenerator generate the token